### PR TITLE
Try add oddjobs

### PR DESCRIPTION
**Pull request by jappeace**
WIP #110 

## Contributor checklist

- [x] My PR is related to https://github.com/flora-pm/flora-server/issues/110
- [x] I have read and understood the [CONTRIBUTING guide](https://github.com/flora-pm/flora-server/blob/development/CONTRIBUTING.md)
- [x] I have inserted my change and a link to this PR in the [CHANGELOG](https://github.com/flora-pm/flora-server/blob/development/CHANGELOG.md)

before
image

after
image

### CODE REVIEW

File: src/Flora/Environment.hs [hunk 1]
```diff
@@ -1,44 +1,29 @@
-{-# OPTIONS_GHC -Wno-orphans #-}
+{-# LANGUAGE PackageImports #-}
```
Comment (tchoutri) [inline src/Flora/Environment.hs:1]: Is this still required?

File: src/Flora/Environment/OddJobs.hs [hunk 2]
```diff
@@ -0,0 +1,60 @@
+{-# LANGUAGE PackageImports #-}
```
Comment (tchoutri) [inline src/Flora/Environment/OddJobs.hs:1]: Is this still needed?

File: src/Flora/Environment/OddJobs.hs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+{-# OPTIONS_GHC -Wno-orphans -Wno-deprecations #-}
```
Comment (tchoutri) [inline src/Flora/Environment/OddJobs.hs:2]: What deprecation are we avoiding here?

File: src/Flora/Model/PersistentSession.hs [hunk 4]
```diff
@@ -1,9 +1,11 @@
+{-# LANGUAGE PackageImports #-}
```
Comment (tchoutri) [inline src/Flora/Model/PersistentSession.hs:1]: Is this still needed?

File: cabal.project.freeze [hunk 5]
```diff
@@ -206,7 +206,6 @@ constraints: any.Cabal ==3.6.3.0,
              any.random ==1.2.1,
              any.raven-haskell ==0.1.4.1,
              any.regex-applicative ==0.3.4,
-             any.resource-pool ==0.3.0.0,
```
[the remark this answers is not in the document]
Comment (jappeace) [reply cabal.project.freeze]: all these package imports and resorce-pool things were due to me hacking cabal into supporting 2 resource pools.
but now resource-pool 3 is released I patched oddjobs, so it's no longer necessary (and becomes a lot simpler).
I put this back.

File: src/Flora/Environment.hs (hunk 1 above)
Comment (jappeace) [reply src/Flora/Environment.hs]: I deleted all of these

File: migrations/20220531232510_create_oddjobs.sql [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+    created_at timestamp with time zone DEFAULT now() NOT NULL,
+    updated_at timestamp with time zone DEFAULT now() NOT NULL,
+    run_at timestamp with time zone DEFAULT now() NOT NULL,
+    status text NOT NULL,
+    payload jsonb NOT NULL,
+    last_error jsonb,
+    attempts integer DEFAULT 0 NOT NULL,
+    locked_at timestamp with time zone,
+    locked_by text,
+    CONSTRAINT incorrect_locking_info CHECK ((((status <> 'locked'::text) AND (locked_at IS NULL) AND (locked_by IS NULL)) OR ((status = 'locked'::text) AND (locked_at IS NOT NULL) AND (locked_by IS NOT NULL))))
```
Comment (tchoutri) [inline migrations/20220531232510_create_oddjobs.sql:31]: Could this be split on multiple line? Ideally 80 chars lines but 100 is acceptable.

**Review by tchoutri — CHANGES_REQUESTED** [review]
:+1:

File: cabal.project [hunk 7]
```diff
@@ -58,3 +53,16 @@ source-repository-package
     location: https://github.com/scrive/log
     tag: 73b4735
     subdir: ./log-base
+
+-- need to use SupercedeTech until this is merged
+-- https://github.com/saurabhnanda/odd-jobs/pull/88
+source-repository-package
+    type: git
+    location: https://github.com/jappeace/odd-jobs
+    tag: a75515791f2c743614ec05d54493ef12b143002e
```
[the remark this answers is not in the document]
Comment (jappeace) [reply cabal.project]: not quite, they still haven't figured out how to press the merge button on https://github.com/saurabhnanda/odd-jobs/pull/90

File: migrations/20220531232510_create_oddjobs.sql (hunk 6 above)
Comment (jappeace) [reply migrations/20220531232510_create_oddjobs.sql]: yes, I did this.

**Review by tchoutri — APPROVED** [review]

**Comment by tchoutri** [comment]
Thanks a lot @jappeace !!

**Pull request merged**